slick/src/main/scala/slick/util/ReadAheadIterator.scala

```diff
@@ -49,3 +43,11 @@ trait ReadAheadIterator[+T] extends BufferedIterator[T] {
     } else throw new NoSuchElementException("next on empty iterator");
   }
 }
+
+object ReadAheadIterator {
```

Comment:
Extension methods are only searched if the method name is not defined directly on the type (even if the signatures differ). This is exactly what we want here. When used on a Scala version that provides `headOption` on its own, the implicit conversion is never applied.